src/components/Player.ts

```diff
@@ -0,0 +1,102 @@
+import { GameObj, KaboomCtx } from "kaplay";
+import GameUtils from "../game/GameUtils";
+import GameInstance from "../game/Game";
+
+export default class Player { 
+    public static playerObject: GameObj;
+    private static game: KaboomCtx = GameInstance.getGameInstance().getKObject(); 
+
+    private static playerComponents = [
+      Player.game.sprite('bean'),
+      Player.game.area(), 
+      Player.game.body(), 
+      Player.game.pos(800,500),
+      Player.game.anchor('center'), 
+      'player'
+    ]; 
+
+    // constructor() {
+    //     this.playerObject = this.create(); 
+    // }
+
+    public static create() {
+
+      if(this.playerObject) {
+        console.log("PS");
+        
+        return this.playerObject; 
+      }
+
+        //make the object without rendering it to the scene
+        const player = Player.game.make(this.playerComponents); 
+
```

Comment:
move the make function to the Game class and make it take multiple options to make something, hence using that function by only passing options.